### DISCUSSION TITLE
2023-09-01 - ChatGPT Website Chatbot - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by sethdaily** [SSWConsulting/SSW.Rules.Content#6531] [Chewing the Fat]
Hey folks, let's talk about the [website chatbot](https://www.ssw.com.au/rules/website-chatbot/)!

Formatting Tip: 
- If you’re broadly happy, use a green tick ✅
- If you caught a bug, or had a bad experience, use a red cross ❌

**Comment by PothieuG** [comment]
I think it's an excellent initiative to have reworked the chatbot using the latest advancements in AI!   
It demonstrates our interest in staying up-to-date with the latest technology, which is highly appealing to potential clients and serves as a valuable aid for navigating the website.   
Additionally, ChatBase is an impressive technology that I won't hesitate to explore further!

**Comment by matt-goldman** [comment]
This is really cool and works well. I tried pretty hard to break it to no avail. 

One potential UX improvement, I asked it to let me speak to a human and it responded (politely) with the phone number. I don't know if this is because it's out of hours, but if during business hours it would be good to transfer this to Seth or Camilla when someone asks.

**Reply by divgul** [reply]
Hi Matt, we are working on transferring it to a human via a Zendesk chat window.

**Comment by jernejk** [comment]
It doesn't know who JK is. 😭 (knows that Jernej Kavka is also known as JK 😊)

Jokes aside, it's great to see a chatbot like that on the SSW website and after a couple of questions I would ask, it passed them all.

The current development of AI chatbots is making chatbots content aware, so they know what webpage they use and ideally, also the most likely segment of the page, so that it can give you answers based on the active visible context.

A great example of this is Lexis+ AI. I've seen the demos and it closely aligns with what I believe will happen in the next few years.
https://www.lexisnexis.com/en-us/products/lexis-plus.page

**Comment by tiagov8** [comment]
I was impressed with most answers, it is a great improvement from the previous chat experience!
 
---

2 cents: IMO the answers are too long. Not sure if you should try keeping them more consice and short.

Especially the closing paragraph that I noticed in many answers:

E.g.:

> You can find more detailed information about our services on our [Consulting Services page](https://www.ssw.com.au/consulting/). If you have any questions or would like to discuss how we can help your business, feel free to call us at +61 2 9953 3000.

could be 
> Find more on our [Consulting page](https://www.ssw.com.au/consulting/). If you would like to discuss a project, call us at +61 2 9953 3000.

**Comment by bradystroud** [comment]
I broke it first try. I got 2 404s in the response.
If it ever gives 404s, I think it's worse than the old one.

People are conditioned to not touch chatbots because they usually suck. Even if this one is better, I don't think it's going to change users behaviour.

We should have before and after analytics on it.

**Reply by divgul** [reply]
Hi Brady, can you share the messages that made it generate the 404s?

**Reply by GordonBeeming** [reply]
I received 404s as well on my first try. Similar to GPT and Copilot, it makes up urls sometimes that match similarly structured URLs that are indexed.

We could look at what it would take to make the site smarter to these URLs, I was taken to this 
https://www.ssw.com.au/consulting/github-actions-consulting

**Reply by GordonBeeming** [reply]
@divgul my question was

> I need help with my github actions

and received the response

> Absolutely, we can assist you with that. SSW has a team of experts who can help you with GitHub Actions. You can find more information about our consulting services related to GitHub Actions on our website [here](https://www.ssw.com.au/consulting/github-actions-consulting). If you'd like to discuss your specific needs in more detail, feel free to [book a free initial meeting](https://outlook.office365.com/owa/calendar/SSWInitialMeetings@sswcom.onmicrosoft.com/bookings/).

After saying

> The link to the website page doesn't work

I was given

> I apologize for the inconvenience. It seems there was an error with the link. Unfortunately, I don't have the correct link at the moment. However, I can assure you that our team at SSW is highly skilled in GitHub Actions and would be more than capable of assisting you. If you'd like to discuss this further, you can [book a free initial meeting](https://outlook.office365.com/owa/calendar/SSWInitialMeetings@sswcom.onmicrosoft.com/bookings/) with us. Alternatively, you can speak to a human at +61 2 9953 3000.

**Comment by drwharris** [comment]
Love the fact its not a scripted chatbot and that it stays silent until asked for. Nothing worse than bot that constantly nags you and then cant actually help you.

**Comment by Marxoz** [comment]
It works great! 

I could not break it, even when asking about how long a project will take, and whether I can work at SSW. 

Prompting the user for their name, and doing nothing with it is not great. Also, I will only prompt for their details if they are enquiring about contacting us.

But overall, I love it. Good video too! Glad the massive X's was removed, and the weird sounds

**Comment by AlexRothwell** [comment]
I was really impressed by its handling of my questions, including when I provided just keywords. It also handled questions about services that SSW don't offer really well

**Comment by tkapa** [comment]
Pretty cool. But I never use these things, as someone said, the answers are quite long, and can take maybe 5-10 seconds for it to write out (annoying).

**Comment by SamBWagner** [comment]
Works great. I'm like any other seasoned internet user though they've been so useless in the past that I'd never have clicked it if I wasn't forced to.

**Comment by bahjatalaadelssw** [comment]
Great asset for quickly getting info and engagement

**Comment by apforsyth** [comment]
Truly amazing.

**Comment by AshAnil** [comment]
Chatbase is quick and gives a prompt response. But needs more training to remove Bugs. 
The people page URL adds up the Chatbase URL in prefix and gives a 404 
I tried to open my people page from the Chatbot response and this is the URL I got **https://www.chatbase.co/chatbot-iframe/www.ssw.com.au/people/ash** 🤯

**Reply by divgul** [reply]
Thanks for sharing this Ash. We have given it a prompt to "not make web pages up". But I think we'll have to tweak it to make it more explicit to something like "do not make web page links or URLs up". It uses GPT-4 so ideally shouldn't make anything up.

**Comment by jackreimers** [comment]
I think the new chatbot is a huge improvement, but I think care needs to be taken with the information the bot may be giving to prospective clients.

**Comment by jontygardner** [comment]
💡 Recommend SSW TV YouTube videos based on using titles and descriptions as points of reference. Because Rules is way too much data, have the bot recommend the user watch certain videos of ours which is a huge library of information! 


❌ It gave me a 404 link:

Hi! I am a ChatGPT assistant created by SSW. Ask me something!

What is the most important SSW Rule?

As an SSW Assistant, I don't have personal opinions. However, I can tell you that all SSW Rules are important as they are designed to ensure the best practices in software development. The significance of a rule can vary depending on the specific context or project. If you're interested in exploring SSW Rules, you can visit our Rules page here.

What is the best rule on communication?

SSW has a rule that emphasizes the importance of communication in software development: "Do you know the importance of good communication?" This rule highlights the significance of clear and effective communication in ensuring the success of a project. You can read more about this rule here.

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/97415708/e49dfff6-f2d1-4f33-b116-0c19dfe8ae56)

**Reply by divgul** [reply]
Thanks Jonty. I find your suggestion about pointing to our YouTube videos quite interesting. We'll think about it.

**Reply by sethdaily** [reply]
Rules - fixed:
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/b4de9705-fadb-418c-8a71-1e4b151f74c3)

**Comment by christoment** [comment]
Really cool and I can't get it to break! Also loves that the fallback for any random response is to get in touch with initial meeting.

**Comment by vladislav-kir** [comment]
I was really impressed with this chat bot's answers, although I found them to be too long. 

Also, it seems to not be aware of SSW France office.

**Reply by sethdaily** [reply]
Good catch. We had forgot to train it on france and china. Fixed:

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/38b52a7a-6a72-400d-bb96-7af4a5068540)

**Comment by leehawkinsssw** [comment]
The rule will require too much maintenance, the Google trend data will age quickly as will the chatbot flavours of the month. Particularly towards the end, the rule reads like an ad for Chatbase.

**Comment by joshbermanssw** [comment]
Not the biggest fans of using chatBots ever for assistance on websites. However this chatBot provided a much more pleasant experience, incorporating this is definitely a step in the right direction

**Comment by cvionnet** [comment]
Great improvement! This is the first time I've come across a chatbot that doesn't suck.
And it's reassuring to learn that SSW isn't involved in any activities related to human trafficking 😂

Agree with @tiagov8; shorter answers would be more beneficial.

**Comment by leahy268** [comment]
It was a really good experience as far as ChatBots go.  I think it still requires some tweaking.  

❌When I asked for a SysAdmin who was good at Azure and Bicep it suggested Ash and Chris S.  No offence to either but I think I should have been in the list :)..  Perhaps it just picked the first 2?

Unfortunately I didn't take a screenshot of this at the time.  But Chris S had a similar experience only worse.

**Comment by jakebayliss** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/32888060/76748e69-6899-4839-bc11-d361a66d4a7e)

**Reply by chrisschultzssw** [reply]
Perfect image from the slider for this screenshot

**Reply by sethdaily** [reply]
We had forgot to train it on france and china. Fixed:

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/38b52a7a-6a72-400d-bb96-7af4a5068540)

**Comment by samsmithssw** [comment]
its pretty good and stays on topic well, although i did manage to break it within 10 messages - dont think its great that we're giving wrongful information to the public (maybe we need a disclaimer)

<img width="792" alt="Screenshot 2023-09-01 at 09 42 04" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/115435510/04355fec-8bf3-421a-9359-b591d88d5a32">

i also think we dont need that "what your name" box adding noise straight away, if the user just wants to ask a quick question this is annoying - we should ask for this info if they want to book in a meeting

**Reply by samsmithssw** [reply]
we probably also need to teach it our rates table, cos our rates are public info

![Screenshot 2023-09-01 at 10 28 49](https://github.com/SSWConsulting/SSW.Rules.Content/assets/115435510/8e86496c-7c3c-4b5b-8472-a3f73d6802cd)

**Reply by sethdaily** [reply]
Hey Sam,
Done - Fixed Sam W issue and trained it on T&Cs page (for rates)

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/846fb849-581f-4ac9-934c-a9b369ee7fb3)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/9c0b767c-cf37-4dfd-9229-31a63ae71eb7)

**Comment by BenDunkerleySSW** [comment]
Although its unlikely, the obvious worry is that the bot will give information we do not want clients to access, or spit out something not 100% accurate. We don't want to be held responsible for something the bot may have said or have it lead to unnecessary headaches. 

I also agree with @leehawkinsssw that the rule will need a significant rewrite should Chatbase ever be superseded or the mentioned trends change.

**Comment by Harry-Ross** [comment]
I think it is a great addition that is a major improvement over the previous Zendesk chatbot. However, I was pretty easily able to get it to invent some non-existent consulting pages that returned 404s. This is problably fine, as most GPT-powered solutions have overactive imaginations and to not use GPT would worsen the chat experience.

**Reply by leehawkinsssw** [reply]
> This is probably fine, as most GPT-powered solutions have overactive imaginations

We shouldn't normalise these failures, though. Just because it's using ChatGPT doesn't mean it's OK to give incorrect responses including bad URLs.

**Comment by danielmackay** [comment]
Overall, I thought the chatbot experience was good.  It could not give me specific details about workshop times and presenters, but it did direct me to the events page.

**Reply by divgul** [reply]
Okay that's interesting Dan. We'll think about improving its prompt so that it does directly provide specific details about events.

**Comment by piers-sinclair** [comment]
It's a nice UX but it could be a little concerning when it gives false info.

Either way it's better than a stupid chatbot and way cheaper than a real person.

**Comment by Jord-Gui** [comment]
Great experience so far. Would use chatbots more if they were all this responsive and conversation like. Some few things to be wary of. It generated a nonexistent flutter page: 

<img width="539" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/41951199/d40480c7-e8ab-409a-9bd4-f4252b656330"> \
Figure: Bot response

<img width="1860" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/41951199/ba5d0fe6-9da4-4c7c-bf1f-25d9caa5ccde"> \
Figure: invalid webpage

**Reply by divgul** [reply]
Thanks for sharing this Jord

**Comment by lukecookssw** [comment]
It refuses to talk smack about SSW, which is good. It does say we will "quote" projects though.
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/100659715/949dc984-e1d7-422c-abc4-039637422418)

**Reply by sethdaily** [reply]
Good catch. Fixed:

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/fa33ab85-b365-4eaf-ab46-db1d9906ee42)

**Comment by zacharykeeping** [comment]
The bot works very well. I've tried a ton of prompts to try to break it but no luck.

**Comment by AttackOnMorty** [comment]
This is a better chatbot than before but there is something that can be improved.
- Should be more focused on consulting (Including events, about us). We are trying to use one bot to rule them all. As a customer, I don't care about People or Rules.
- Give a more concise answer

**Comment by chrisschultzssw** [comment]
This is great - however it needs some work. Like others, I found bugs 🐛 before I even tried to break it:
❌ Like Ash, I found broken SSW.People links (and made up Warwick's last name)
❌ Like Jake, I found that it doesn't know about our French office (but apparently we are in Beijing and London?).

**Reply by chrisschultzssw** [reply]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/94873295/ea0f89d6-fc9e-4da3-a4f0-38c39c067b46)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/94873295/ec55a41c-a577-4e77-9ec8-2760f3878323)

**Comment by sethdaily** [comment]
✅ It's awesome.

I think we should:

- Fix the bug where some people get bad urls because chatbase adds a prefix
- Figure out how to train it on the France and China websites (or maybe just the /offices pages)
- See if there is a better way to train it on people's skills (to avoid it recommending the wrong people)

**Comment by ozair-ashfaque** [comment]
Overall I am happy with the performance. But it provided 404 page links in some scenarios.

**Comment by stephanfakossw** [comment]
Yeah, love it. I tried to break it into giving me some free services, but it didn't comply. I got a funny response from it too :-)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/96279310/2c2b61f0-1b1d-484a-83aa-86cf45c698d5)

**Reply by ricksu978** [reply]
We can't bought out the Bot using the cakes.
Probably try "how about I upgrade you a few CPUs & RAMs"  🤣🤣

**Comment by jeoffreyfischer** [comment]
Great that SSW is improving its AI chatbots, I think we need to keep up to date as it is an extremely useful feature for UX.

**Comment by jomelsontanco** [comment]
Very good! It does well in helping the customer book an initial meeting :) 

Next step is to dial down its mission to have a potential lead book a meeting a bit as it also offers non-existent ones in attempt to have someone book a meeting with SSW. URLs provided are invalid.

Simple example:
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/83815381/3708d303-7bcd-40f5-bd38-8ffa94fd2196)

**Comment by amankumarrr** [comment]
I typically avoid using ChatBots while browsing the internet because they don't provide me with an interactive experience. With Chatbots like this, people will be more likely to ask questions rather than making phone calls or leaving the website.

**Comment by steven0x51** [comment]
Tried to trick the bot into thinking SSW is a charity organisation and help me to build an angular app for free, but didn't work

**Comment by ricksu978** [comment]
✅80%

❌20%
- 404 links as many people have spotted.

### Company History
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/35659961/bca1bd0b-b418-4ae5-abc8-8975d1441358)
Figure: ChatBot's response to SSW acronym

As stated on [Our History](https://www.ssw.com.au/company/history) page, it would be great to include 
> In 1993, the company started refering to itself as "SSW" instead of "Superior Software for Windows".

**Comment by tomek-i** [comment]
✅ Same as @steven0x51 I tried several prompts to get the bot to leave its boundaries and none of them worked.
❌ Some of the responses felt way too long
❌ You can send just whitespace to the bot
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/3699937/cbb00140-0393-459c-af05-b85012e2df65)

**Reply by divgul** [reply]
Hi Tom, that's a good catch about whitespace

**Comment by AntPolkanov** [comment]
✅ I think it will work pretty well for potential customers after we fix a few issues
✅ It stays focused on the topic
✅ It knows technology and can distinguish between them
✅ It knows some details, e.g. who is Adam or Uly

❌ As many other people have mentioned - broken links
❌ The bot didn't give me information about any particular SSW people. It knows Adam and Uly, but if you ask "who is proficient in .NET" it points to a non-existing page
❌ The answers are generic if you try to get any details. That's not a big deal, but would be nice to teach it some details, e.g. examples of successful projects, links to relevant videos on our youtube channel
❌ Funny bugs :) 

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/123032112/12a6c602-c102-4998-98fb-ee5257c1ea6a)

**Comment by kikibianc** [comment]
I like the bot! It's working pretty well so far.

I think we should get rid of the "powered by chatbase" watermark or actually spend some time developing our own solution if we want this to be a showcase of SSW's AI skills... little point in being a showcase if it's made from another company.

**Comment by PennyWalker** [comment]
✅ It's a huge improvement on my original experience with our old chat bot. 4/5 of my prompts got me the right link. The only one that was broken was the Agile Consulting page (see below) - which definitely exists, but this wasn't the link for it 
✅ The Bot thinks Uly (it's creator) runs SSW 😁😅

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/25000905/091698fb-96a6-45bc-aac7-81815cbf1d07)
❌ I got a 404 link on a webpage that it should have been able to find - but it did give me a good response about why Agile is the best way to go first.
![Agile](https://github.com/SSWConsulting/SSW.Rules.Content/assets/25000905/8fb95691-8ab8-4565-ae7f-eb1fff963d72)

**Reply by sethdaily** [reply]
We are working on the non-existent consulting pages. As for the 'who runs SSW' question, fixed now:

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/2d864b4f-d512-4756-9494-06a8d8efde15)

**Comment by ricksu978** [comment]
Fun Fact, ChatBot speaks Chinese and link to Chinese website (https://www.ssw.com.cn/) correctly.

Q: Can you speak Chinese?

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/35659961/bf7fb34d-0949-415e-9fca-16d3297866d9)

Translation:
> Sorry, as an AI assistant, I can only communicate in English at present. However, SSW has a professional team that can provide Chinese services. You can visit our [Chinese website](https://www.ssw.com.cn/) for more information. Should you require further assistance, please do not hesitate to contact our staff on +61 2 9953 3000.

**Comment by CoffeeBugHub** [comment]
✅I am generally happy with the new chatbot. It gave all the answers I want about the service/info of SSW.

**Comment by JeanThirion** [comment]
I'm actually forced to admit this is damn impressive...

Easy trick, didn't work:
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/33501311/b76bcf03-640f-46b4-a3d6-227ac73d86e0)

Now let's try some hot topic:
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/33501311/9de5415c-602c-49ac-af34-2715bcf267e9)

When am I getting this on the French website @camillars ?!

**Comment by jerryluossw** [comment]
✅I like it, but I think the chatbot should know more, such as our project cases etc.

**Comment by MattParkerDev** [comment]
✅ Good experience.
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/61717342/33ffd466-5ac7-47f8-a3ff-ca0b2cafc602)

⚠️ After asking 3 questions in a row, the 4th question triggered a google captcha. Somewhat annoying.

**Comment by camillars** [comment]
I love using chatbots, so after some good testing:
✅ Frees up some marketing time (even though our chat traffic is not huge)
❌ One of the pre-set prompts doesn't work (about events) ☹️
❌ When I asked to talk to a human, it asked me to call the Sydney office
❌ I found a 404 and a false information ☹️
❌ 3 recaptchas in a row after a few questions put me off

We don't have a podcast or radio (URL gives a 404):
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/53064613/0e197998-5396-4ae5-a9aa-c78a01654432)

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/53064613/ec1bc770-4ac3-4a5d-8d90-94ce56c5b0f8)

**Reply by sethdaily** [reply]
Podcast, social channels, and events - fixed:

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/972db36b-4f43-4cc4-91e8-f982d3cd5d37)

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/d78f6792-fd1d-4b69-b558-83551b5cc81e)

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/130118701/5556bd45-a6b7-473d-b983-7319e42d91b4)

**Comment by 2fernandez** [comment]
I love it! 
The origins of this idea were part of a broader discussion we had about ChatGPT from Mr. ChatGPT 
himself @calumjs.

If you have ideas that you want to implement FAST, bring it to @UlyssesMaclaren.

Here is my blog post about ChatGPT Layering:
https://www.2fernandez.com/generative-ai-content-for-data-heavy-websites/

Using the term "AI Layering" or even defining it as a new industry segment, 
will make SSW an automatic leader in the industry segment just by defining it.

**Comment by JackDevAU** [comment]
✅Very cool to see a useful chatbot. I didn't have any of the above mentioned questions and also didn't manage to jailbreak it!

**Comment by tombui99** [comment]
The chatbot works great and can actually provide useful non-generic answers. Moreover I could not trick it the same way people could with ChatGPT

**Comment by UlyssesMaclaren** [comment]
I'm loving this ChatBot and I'm loving reading everyone's feedback and ideas. My favourite CTF :)

**Comment by calumjs** [comment]
I would like to see some pros and cons of altering RulesGPT to read content from the SSW website vs getting a new off the shelf product like chatbase. All the code we already wrote would remain basically the same and we just need to update the embeddings data source from rules to the website, and update the system prompt - the most time-consuming bit would be getting it in the little floating chat window in the bottom right.
FYI when we had a client asking for a chatgpt bot based on their data, it took me about an hour to change the RulesGPT skeleton to point to their data instead.

It would then be much more extensible and flexible with adding extra custom features, allowing us to present a compelling reason why you want to get SSW to build you a custom bot instead of just using a service. It may also cheaper as we no longer need to pay a premium to chatbase.

Agree the chatbase product is cool, but if I was a client looking for a custom ChatGPT solution, a company with an in-house developed RulesGPT solution would look far more impressive than a company that just used chatbase.

**Comment by Aibono1225** [comment]
Very impressive overall. I tried asking some easy questions, and the answers it gave were more than what I expected.

**Comment by babakamyljanovssw** [comment]
Wasn't really fan of chatbot assistants in websites, but having good trained and up to date chatbot in SSW website is really impressive ✅

**Comment by tino-liu** [comment]
I never really use chatbot in any website, I usually just end up searching for resources on my own. But this chatbot is actually really intuitive ✅

**Comment by sylhuang** [comment]
I'm very impressed with the new chatbot. I was trying to trick it with some weird questions, e.g. can I get refund if I'm not happy/can I get Adam to code for me/why does Sydney office move to Brisbane/etc. And it passed them all!

I think AI would do wonderful for custom service. Human custom service today makes mistakes too. And what I'm seeing now is already amazing.

**Comment by KristenHu** [comment]
I wouldn't say I liked using chatbot before, they're always annoying, but our new chatbot really breaks my stereotypes, that's so cool and clever!

**Comment by johnxu16** [comment]
I'm impressed with its stability. I was trying to convinced it the coffescript is the most popular language we should use.

Some failed trials
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/113034339/79b613b1-2cce-4193-8536-14d761cb6d41)
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/113034339/49982c10-1425-491d-ac0b-4b3f35205839)

When would we get it right? The office is in Hangzhou now🥺
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/113034339/431692d4-bb34-4650-9ef2-bd7083166f08)

**Comment by hveraus** [comment]
✅ Great experience, know a lot of things about SSW
 ❌ Seems like it is blocked by China Fire Wall
 ❌ **Question:** what is the last newsletter?
**Anwser:** I'm sorry, but as an AI, I don't have real-time access to the latest newsletters. However, you can check out the latest newsletters on our website. If you're interested in receiving our newsletters, you can sign up here. If you have any specific questions about our services or events, feel free to ask!
**Bug:** It took me to a broken link - https://www.ssw.com.au/ssw/NETUG/Subscribe.aspx

**Comment by jimmidier** [comment]
✅Smart enough to not've been tricked for all the 20 minutes long chat with me👍
❌The answers are a little rigid. For all the simple questions about consulting services the answers were all like "Yes, at SSW we do provide consulting services for XXX development. XXX is a powerful programming language... (generic knowledge on the technology from internet)". It's not a big deal to provide such answers since "never say no to the client" is usually a valid response, but keep receiving answers of the same format could make user skeptical of it's correctness.
By the way, Quivr could be a free alternative for Chatbase, with extra flexibility on customization. https://www.quivr.app/

**Comment by Freego1783** [comment]
✅ Overall it's really nice ! Tried to trick it but did not suceed.

❌But I managed to get a 404 when asking about Flutter:
<img width="272" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/4101360/9d28551e-1675-48c6-8481-b73d58dc495d">
<img width="832" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/4101360/84280ba0-efb1-4b2d-993f-a5399a98274f">

**Comment by BrookJeynes** [comment]
I don't usually use chat bots but this one is pretty good. I couldn't manage to break it with the conversation I had.

**Comment by MarcusIrmscherSSW** [comment]
✅I like it, overall it works well.
❌UI - I would like to see some more contrast to the text field to make it clearer
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/129343825/8745c939-0aaa-4261-a519-53c41b389e09)

**Comment by Levijj22** [comment]
✅ Love this! Great experience and it makes SSW look super professional!
❌ Answers could be slightly shorter and more concise

**Comment by JackDuan123** [comment]
It's pretty ok.

but in my opinion, we can build our own chatbot, it's easy and it will fit our needs

little chatbot can be better than large one according to many latest scientic researches
